enhancing/enhancer.spec.js

```diff
@@ -1,2 +1,125 @@
+    });
+  });
+  describe("Succeed test suite", () => {
+    it("should increase the enhancement by one if less than 20", () => {
+      // arrange
+      const item = { name: "broadsword", enhancement: 0, durability: 80 };
+      const expected = {
+        name: "broadsword",
+        enhancement: 1,
+        durability: 80
+      };
+      // act
+      const actual = enhancer.succeed(item);
+      // assert
+      expect(actual).toEqual(expected);
+    });
+    it("should not change the enhancement level if it is 20", () => {
+      // arrange
+      const item = { name: "broadsword", enhancement: 20, durability: 80 };
+      const expected = {
+        name: "broadsword",
+        enhancement: 20,
+        durability: 80
+      };
+      // act
+      const actual = enhancer.succeed(item);
+      // assert
+      expect(actual).toEqual(expected);
+    });
+  });
+  describe("fail test suite", () => {
+    it("should decrease the durability of an item by 5 if the enhancement is less than 15", () => {
+      // arrange
+      const item = { name: "broadsword", enhancement: 14, durability: 80 };
+      const expected = {
+        name: "broadsword",
+        enhancement: 14,
+        durability: 75
+      };
+      // act
+      const actual = enhancer.fail(item);
+      // assert
+      expect(actual).toEqual(expected);
+    });
+    it("should decrease the durability of the item by 10 if the enhancement is 15 or more", () => {
+      // arrange
+      const item = { name: "broadsword", enhancement: 16, durability: 80 };
+      const expected = {
+        name: "broadsword",
+        enhancement: 16,
+        durability: 70
+      };
+      // act
+      const actual = enhancer.fail(item);
+      // assert
+      expect(actual).toEqual(expected);
+    });
+    it("decreases the enhancement level by one if the enhancement level is greater than 16 and durability by 10", () => {
+      // arrange
+      const item = { name: "broadsword", enhancement: 17, durability: 80 };
+      const expected = {
+        name: "broadsword",
+        enhancement: 16,
+        durability: 70
+      };
+      // act
+      const actual = enhancer.fail(item);
+      // assert
+      expect(actual).toEqual(expected);
+    });
+  });
+  describe("get tests suite", () => {
+    it("should not modify the name at all if the enhancement level is zero", () => {
+      const item = { name: "broadsword", enhancement: 0, durability: 80 };
+      const expected = {
+        name: "broadsword",
+        enhancement: 0,
+        durability: 80
+      };
+      // act
+      const actual = enhancer.get(item);
+      // assert
+      expect(actual).toEqual(expected);
+    });
+    it("if enhancement is greater than zero should add [+n] to the name where n is the ehnhancement level  ", () => {
+      const item = { name: "broadsword", enhancement: 1, durability: 80 };
+      const expected = {
+        name: "[+1] broadsword",
+        enhancement: 1,
+        durability: 80
```

Comment:
You could also alter the main helper for this sort of test:
```javascript
const makeItemMaker = name => ({
  e: enhancement = "0", d: durability = 100, m: modifier
}) => ({
  name: modifier ? `[${modifier}] ${name}` : name,
  enhancement.
  durability
});
```
and the usage, assuming an already declared makeBroadsword helper (which could be declared up a scope):
```javascript
const item = makeBroadsword({ e: 1, d: 80 });
const expected = makeBroadsword({ e: 1, d: 80, m: '+1' });
```